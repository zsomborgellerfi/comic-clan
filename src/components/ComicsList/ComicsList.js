import React from "react";
import ComicsListItem from "../ComicsListItem";
import { Row, Col } from "react-bootstrap";

import "./ComicsList.scss";

const ComicsList = ({ comics }) => {
  return (
    <Row>
      {comics.map((comic, idx) => (
        <Col key={idx} className={"ComicsList__col"} xs={12}>
          <ComicsListItem comic={comic} />
        </Col>
      ))}
    </Row>
  );
};

export default ComicsList;
