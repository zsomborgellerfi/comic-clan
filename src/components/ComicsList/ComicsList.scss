@import '../../styles/variables';

.ComicsList {
  &__container {
    display: flex;
    flex-direction: column;
  }
  &__col {
    max-width: 20%;
  }
}
