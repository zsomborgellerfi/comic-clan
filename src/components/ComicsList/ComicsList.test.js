import React from "react";
import { shallow } from "enzyme";
import ComicsList from "./ComicsList";

const props = {
  comics: [{ name: "test comic 1" }, { name: "test comic 2" }],
};
describe("ComicsList", () => {
  const wrapper = shallow(<ComicsList {...props} />);

  it("should render 2 ComicsListItem correctly", () => {
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("ComicsListItem")).toHaveLength(2);
  });
});
