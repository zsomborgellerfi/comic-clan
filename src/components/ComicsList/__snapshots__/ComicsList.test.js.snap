// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComicsList should render 2 ComicsListItem correctly 1`] = `
<Row
  noGutters={false}
>
  <Col
    className="ComicsList__col"
    key="0"
    xs={12}
  >
    <ComicsListItem
      comic={
        Object {
          "name": "test comic 1",
        }
      }
    />
  </Col>
  <Col
    className="ComicsList__col"
    key="1"
    xs={12}
  >
    <ComicsListItem
      comic={
        Object {
          "name": "test comic 2",
        }
      }
    />
  </Col>
</Row>
`;
